lib/rust/parser/src/syntax/tree.rs

```diff
@@ -755,6 +757,50 @@ impl<'s> span::Builder<'s> for OperatorDelimitedTree<'s> {
 
 
 
+// ================
+// === Warnings ===
+// ================
+
+/// Warnings applicable to a [`Tree`].
+pub type Warnings = ColdVec<Warning>;
+
+
+// === Warning ===
+
+/// A warning associated with a [`Tree`].
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Reflect, Deserialize)]
+#[allow(missing_copy_implementations)] // Future warnings may have attached information.
+pub struct Warning {
+    id: u32,
```

Comment:
Why we don't use just WarningId here (And make WarningId `#[repr(u32)]`)?

Reply:
This representation is designed for FFI. If we used the parser in Rust I would use an enum with parameters, with serialization overridden to produce this representation. Since the Rust API is only used by debug tools, I cut out the Rust-friendly type.